extensions/arc/src/test/ui/components/radioOptionsGroup.test.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as should from 'should';
+import * as sinon from 'sinon';
+import { getErrorMessage } from '../../../common/utils';
+import { RadioOptionsGroup, RadioOptionsInfo } from '../../../ui/components/radioOptionsGroup';
+import { loadingError, modelView, radioButtons } from '../../mocks/fakeContainersAndBuilders';
+
+const radioOptionsInfo = <RadioOptionsInfo>{
+	values: [
+		'value1',
+		'value2'
+	],
+	defaultValue: 'value2'
+};
+
+let radioOptionsGroup: RadioOptionsGroup;
+
+describe('radioOptionsGroup', function (): void {
+	beforeEach(async () => {
+		radioOptionsGroup = new RadioOptionsGroup(modelView, (_disposable) => { });
+		await radioOptionsGroup.load(async () => radioOptionsInfo);
+	});
+
+	afterEach(() => {
+		sinon.restore();
+	});
+
+	it('verify construction and load', async () => {
+		should(radioOptionsGroup).not.be.undefined();
+		should(radioOptionsGroup.value).not.be.undefined();
+		radioOptionsGroup.value!.should.equal('value2');
+		// verify all the radioButtons created in the group
+		verifyRadioGroup('value2');
+	});
+
+	it('onClick', async () => {
+		//click the radioButton corresponding to 'value1'
+		radioButtons.filter(r => r.value === 'value1').pop()!.click();
```

Comment:
Yeah so tying the specific test implementation of the object creation together like this isn't great. You should be having generic functions that create the objects but then leave it up to the logic you're testing to actually hook stuff up and just have the specific tests do the checks to make sure that things were added as expected.

Reply:
So, I am not sure I am following your comment in the context of the code here. We are just clicking the 2nd radio option within the test radio group to test what out that the radio group's state changes correctly.

Reply:
No change here, can you give a rough sample of what you are looking to happen here ...